src/scheduler/job_info.c

```diff
@@ -1314,9 +1313,15 @@ query_job(struct batch_status *job, server_info *sinfo, schd_error *err)
 		/* array_indices_remaining */
 		else if (!strcmp(attrp->name, ATTR_array_indices_remaining))
 			resresv->job->queued_subjobs = range_parse(attrp->value);
-		else if (!strcmp(attrp->name, ATTR_execvnode))
-			execvnode = attrp->value;
-		else if (!strcmp(attrp->name, ATTR_l)) { /* resources requested*/
+		else if (!strcmp(attrp->name, ATTR_execvnode)) {
+			nspec **tmp_nspec_arr;
+			tmp_nspec_arr = parse_execvnode(attrp->value, sinfo, resresv->select);
+			resresv->nspec_arr = combine_nspec_array(tmp_nspec_arr);
+			free_nspecs(tmp_nspec_arr);
+
+			if (resresv->nspec_arr != NULL)
+				resresv->ninfo_arr = create_node_array_from_nspec(resresv->nspec_arr);
```

Comment:
The reason this was at the bottom was the fact that resresv->select might not be queried at this point.  In any case, you probably don't need to worry about that because you don't need to map nspecs to chunks.  The chunk mappings are discarded in combine_nspec_array().  Just pass NULL instead of resresv->select.